### PR TITLE
docs/development: create `errors-diagnostic.md`

### PR DESCRIPTION
**Pull request by tigercosmos**
r? @poyenc @a127a127

### CODE REVIEW

File: docs/development/errors-diagnostic.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+### Registering a New Diagnostic Error
+
+To register a new diagnostic error, go to [`include/onnc/Diagnostics/DiagCommonKinds.inc`](../../include/onnc/Diagnostic/DiagCommonKinds.inc) and create a new `DIAG`.
+
+`DIAG`'s format is `DIAG(error name, error level, message)` where `message` is a string with placeholders `%N`, starts from `%0`, `%1` and so on, that receives inputs.
+
+For example, if a new error, `numerical_error`, is going to be registered. The line should be added to the end of the file `DiagCommonKinds.inc`:
+
+```inc
```
Comment (poyenc) [inline docs/development/errors-diagnostic.md]: should the style be `cpp` too?

File: docs/development/errors-diagnostic.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+DIAG(numerical_type_error, Fatal, "The number is %0 type where it should be %1 type.")
```
Comment (poyenc) [inline docs/development/errors-diagnostic.md]: `numerical_error` or `numerical_type_error`?

File: docs/development/errors-diagnostic.md (hunk 1 above)
Comment (tigercosmos) [reply docs/development/errors-diagnostic.md]: it's ok here, since Github can read the type.

File: docs/development/errors-diagnostic.md (hunk 2 above)
Comment (tigercosmos) [reply docs/development/errors-diagnostic.md]: should be `numerical_type_error`. fixed

**Pull request closed without merging**